pkg/webhook/policy.go

```diff
@@ -246,23 +255,49 @@ func (h *validationHandler) getDenyMessages(res []*rtypes.Result, req admission.
 					logging.ResourceName:         resourceName,
 					logging.RequestUsername:      req.AdmissionRequest.UserInfo.Username,
 				}
-				eventMsg := "Admission webhook \"validation.gatekeeper.sh\" denied request"
-				reason := "FailedAdmission"
-				if r.EnforcementAction == "dryrun" {
+				var eventMsg, reason string
+				switch r.EnforcementAction {
+				case string(util.Dryrun):
 					eventMsg = "Dryrun violation"
 					reason = "DryrunViolation"
+				case string(util.Warn):
+					eventMsg = "Admission webhook \"validation.gatekeeper.sh\" would have denied the request"
```

Comment:
"would have denied the request" seems more appropriate for dryrun, where we are presumably dry-running against a future deny. The request "would have been rejected" if this had not been a dry run.

"raised a warning for this request" maybe?

Reply:
updated `eventMsg`. should we change warning message to include `raised a warning for this request` too?

so it's either
`Warning: [would have denied by cm-must-have-gk] you must provide labels: {"gatekeeper"}`

or 

`Warning: [cm-must-have-gk raised a warning for this request] you must provide labels: {"gatekeeper"}`

Reply:
Why not just:

`Warning: [cm-must-have-gk] you must provide labels: {"gatekeeper"}`

?

It sounds like kubectl is responsible for providing the context about what is a warning vs. not?

Reply:
Ah, I see we're adding "denied by" by default.

We could use `[warning from cm-must-have-gk] .....`

Reply:
`Warning:` in the beginning is automatically added, so that'll be 

`Warning: [warning from cm-must-have-gk] you must provide labels: {"gatekeeper"}`

Reply:
Is it added for each line if there are multiple violations?

Reply:
yes

```
Warning: [constraint1] ...
Warning: [constraint2] ...
```

Reply:
dropping the extraneous text for both rejections and warnings looks cleaner to me.

Reply:
Remove both `denied by ...` and `would have denied by ...`?

for 1 warning, 1 deny:
```
Warning: [cm-must-have-gk] you must provide labels: {"gatekeeper"}
Error from server ([deny-all-deny] DATA: {}): error when creating "test/bats/tests/bad/bad_cm.yaml": admission webhook "validation.gatekeeper.sh" denied the request: [deny-all-deny] DATA: {}
```

Reply:
I like that approach, personally.

Reply:
Wouldn't this change existing behavior/expectation though? especially if users have been parsing that string

@ritazh wdyt?

Reply:
It would change existing behavior. I suppose whether that matters depends on whether we view the string format as part of our API. I'd consider these intended to be human-readable, and therefore not part of the API.

HTTP status codes and the API as defined by K8s should be what machines rely on.

Reply:
updated